src/testdrive/src/action/kafka/add_partitions.rs

```diff
@@ -46,10 +47,12 @@ pub async fn run_add_partitions(
             }
         }
         None => {
-            bail!(
+            if !allow_changing_unmanaged_topic {
```

Comment:
I think this entire check branch should be removed , which would simplify the code. The code does not seem to be tracking anything about topics at all, so there should be no distinction between "managed" and "unmanaged" topics at that stage.